src/main/java/io/vertx/core/json/JsonArray.java

```diff
@@ -533,7 +533,7 @@ public JsonArray clear() {
    * @return the string encoding
    */
   public String encode() {
-    return Json.CODEC.toString(this, false);
+    return Json.CODEC.toString(this.list, false);
```

Comment:
this seems to be more than a detail, is it a bug ?

Reply:
As I was debugging to figure out the behavior I noticed that we can reduce jackson work, as it internally doesn't need to do a 1st lookup on the registered classes for see that `JsonArray` needs some extra processing. Which is unwrapping the list.

So the behavior didn't change, the stacktraces are just shorter because we help jackson.

Reply:
Same applies to `JsonObject`